docs/doc-system.md

```diff
@@ -0,0 +1,114 @@
+Slang Doc System 
+================
+
+Slang contains a rudimentary documentation generation system. The mechanism used to mark up source is similar to [doxygen](https://www.doxygen.nl/manual/docblocks.html). Namely
+
+```
+/**
+ ... text ... (JavaDoc style)
+ */
+void someFunctionA() {} 
+ 
+/*!
+ .. text .. (QT style)
+ another line
+ */ 
+void someFunctionB() {} 
```

Comment:
My two cents: I think we should make the documentation of the doc system primarily show/encourage a consistent style for how doc comments are written (e.g., only using `/** ... */` and `/// ...`), and then we can document the other available conventions as a later section on something like "syntax compatibility" or something like that.

While the `slang-doc` mode/tool is currently only something we are using internally, the goal would be for it to eventually be something Slang users are encouraged to make use of, and in that case we are responsible or showing them what we think "good" Slang codebases will look like.